packages/trpc/react/trpc.ts

```diff
@@ -43,22 +43,27 @@ export type Endpoint = (typeof ENDPOINTS)[number];
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 const resolveEndpoint = (links: any) => {
-  // TODO: Update our trpc routes so they are more clear.
   // This function parses paths like the following and maps them
   // to the correct API endpoints.
   // - viewer.me - 2 segment paths like this are for logged in requests
   // - viewer.public.i18n - 3 segments paths can be public or authed
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   return (ctx: any) => {
+    // split the segmented path into an array
     const parts = ctx.op.path.split(".");
-    let endpoint;
+
+    // type-safe for ensuring the keys are form ENDPOINTS
```

Comment:
I don't think it works because links is of type any

Reply:
hmm...I guess I wasn't focusing on that part.